src/python_testing/TC_DGWIFI_2_3.py

```diff
@@ -0,0 +1,160 @@
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_DGWIFI_2_3(MatterBaseTest):
+    """
+    [TC-DGWIFI-2.3] Wi-Fi Network Diagnostics Cluster - Command Verification
+
+    This test case verifies the ResetCounts command and subsequent behavior.
+    """
+
+    @staticmethod
+    def is_valid_uint32(value):
+        """Validates a uint32 value."""
+        return isinstance(value, int) and 0 <= value <= 0xFFFFFFFF
+
+    async def send_reset_counts_command(self, endpoint):
+        """Sends the ResetCounts command to the DUT."""
+        cluster = Clusters.Objects.WiFiNetworkDiagnostics
+        await self.send_single_cmd(cluster.Commands.ResetCounts(), endpoint=endpoint)
+
+    async def read_attribute_and_validate(self, endpoint, attribute, validation_func, field_name):
+        """Reads an attribute and validates it using the provided function."""
+        value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=Clusters.Objects.WiFiNetworkDiagnostics, attribute=attribute)
+        if value is not None:
+            asserts.assert_true(isinstance(value, Nullable), f"{field_name} must be of type 'Nullable' when not None.")
+            if value != NullValue:
+                asserts.assert_true(validation_func(value.Value), f"{field_name} has an invalid value: {value.Value}")
+
+    def desc_TC_DGWIFI_2_3(self) -> str:
+        """Returns a description of this test."""
+        return "[TC-DGWIFI-2.3] Wi-Fi Diagnostics Command Verification with Server as DUT"
+
+    def pics_TC_DGWIFI_2_3(self) -> list[str]:
+        return ["DGWIFI.S"]
+
+    def steps_TC_DGWIFI_2_3(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commission DUT to TH (already done)", is_commissioning=True),
+            TestStep(2, "Send ResetCounts command to DUT"),
+            TestStep(2.1, "Verify BeaconLostCount attribute after reset"),
+            TestStep(2.2, "Verify BeaconRxCount attribute after reset"),
+            TestStep(2.3, "Verify PacketMulticastRxCount attribute after reset"),
+            TestStep(2.4, "Verify PacketMulticastTxCount attribute after reset"),
+            TestStep(2.5, "Verify PacketUnicastRxCount attribute after reset"),
+            TestStep(2.6, "Verify PacketUnicastTxCount attribute after reset"),
+        ]
+        return steps
+
+    @async_test_body
+    async def test_TC_DGWIFI_2_3(self):
+        endpoint = self.get_endpoint(default=0)
+
+        # STEP 1: Commission DUT (already done)
+        self.step(1)
+        # Typically, we assume commissioning was performed by harness scripts.
+        attributes = Clusters.WiFiNetworkDiagnostics.Attributes
+
+        # STEP 2: Send ResetCounts command
+        self.step(2)
+        await self.send_reset_counts_command(endpoint)
+
+        # Allow time for the reset to propagate
+        await asyncio.sleep(1)
```

Comment:
is there a way to check if it propagated or not? It would be prefferable to not have sleeps in tests.
If this is not possible, please explain in the comment above why.